app/client/src/services/AuthService.js

```diff
@@ -73,11 +75,13 @@ angular.module('reg')
           });
       };
 
-      authService.createAccount = function(email, password, onSuccess, onFailure) {
+      authService.createAccount = function(email, password, profile, onSuccess, onFailure) {
+        profile.name = profile.firstname + ' ' + profile.lastname;
         return $http
           .post('/auth/register', {
             email: email,
-            password: password
+            password: password,
+            profile: profile
```

Comment:
By the way, is auth/register set up to handle the profile object? Does it use object deconstruction to get the profile? Or does it pass the entire body to the backend controller?

Reply:
auth/register is set up to take the entire profile object on the back-end